examples/training.jl

```diff
@@ -48,34 +48,34 @@ function split_validation(X::AbstractVector, Y::AbstractVector, frac=0.1; rng=Ra
     (X[idxs[1:ntrain]], Y[idxs[1:ntrain]]), (X[idxs[ntrain+1:end]], Y[idxs[ntrain+1:end]])
 end
 
-function train!(loss, ps, data, opt, val_data; n_epochs=10)
+function train!(loss, model, train_data, opt, val_data; n_epochs=10)
     history = Dict(
         "train_acc"=>Float64[], 
         "train_loss"=>Float64[], 
         "val_acc"=>Float64[], 
         "val_loss"=>Float64[]
         )
     for epoch in 1:n_epochs
-        ps = Flux.Params(ps)
-        progress = Progress(length(data); desc="epoch $epoch/$n_epochs")
-        for Xy in data
-            gs = gradient(ps) do
-                loss(Xy)
+        progress = Progress(length(train_data); desc="epoch $epoch/$n_epochs")
+        epoch_loss = 0.0
```

Comment:
This epoch_loss is just left hanging here. But it might be worthwhile calculating it as the mean or sum of the batch losses.

To get the batch loss `val` do:
```
val, grads = Flux.withgradient(model) do m
    loss(m(Xy[1]), Xy[2])
end
```

Reply:
done, added more loss info during progress reporting